pagopa/io-app
# feat: [#176983293] Add a WebView component to support payments 3DS2

## Short description
This PR add a component to handle the pay step during the credit card on-boarding or  on a payment 

**this is how this component works:**
- at the start up, it creates a temporary html hidden form with all pre-set data (props.formFata)
- it appends, at the end of that html, a JS script to auto-submit the created form to the given url (props.postUri)
- it sets that html as starting page
- on each page load request it checks the url
    - if it is the exit url
    - if it contains the outcome code
- when the the exit url is found, it doesn't load it and call the handler props.onFinish passing the found (maybe not) outcome value



### how to test
- `io-dev-server` get the last commit from master, then `yarn install; yarn start`
- include the component `PayWebViewModal` wherever you want as follows
```typescript
<PayWebViewModal
   postUri={"http://127.0.0.1:3000/pay-webview"}
   formData={{ name: "Joe", surname: "Doe", age: "30" }}
   finishPathName={"/payExitUrl/name"}
   onFinish={maybeCode => Alert.alert(maybeCode.toString())}
   outcomeQueryparamName={"code"}
   onGoBack={constNull}
/>
```

you should have this behavior
<img src="https://user-images.githubusercontent.com/822471/108370967-8eb7b200-71fd-11eb-806a-1ce3c94c1f4a.png" height="600"/>


To customize the server redirects and outcome param and value
see https://github.com/pagopa/io-dev-api-server/blob/2b1b232f48564a71713b10b71c3eeb4dfdf1815c/src/routers/public.ts#L56
